src/post.pegjs

```diff
@@ -58,6 +62,34 @@ Token
 HTML_Text
   = [^<]
 
+WP_Block__Image
+  = b:WP_Block_Balanced
+	& { return (
+		  b.blockType === 'image' &&
+			b.children.length <=2 &&
+			b.children[ 0 ].type === 'HTML_Void_Tag' &&
+			b.children[ 0 ].name === 'img'
+	  ) }
+  { return {
+    type: 'WP_Block__Image',
+		src: b.children[ 0 ].attrs.src,
+		caption: b.children[ 1 ]
+	} }
```

Comment:
@nylen this isn't the prettiest but it is declarative and the transformation is explicit. as long as we stick with a straightforward logical expression and don't do anything stateful or complex it's kind of readable.

honestly I don't think this level of parsing is appropriate for `String -> Blocks` but should probably be carried out higher up `String -> GenericBlocks -> SpecificBlocks` just for performance and coupling issues. this would let us introduce a DSL or query language above without messing with the parser or its performance